### PR TITLE
fix(aws-kinesisstreams-kinesisfirehose-s3): fixed to return pattern created s3 bucket

### PR DESCRIPTION
**Pull request by beomseoklee**
*Issue #, if available:*
#133 

*Description of changes:*
1. `aws-kinesisstreams-kinesisfirehose-s3` pattern returns `s3Bucket` and `s3LoggingBucket` if they are created by the pattern.
2. `aws-kinesisstreams-kinesisfirehose-s3` pattern returns `kinesisStreamRole` as `README.md` already described the value.
3. `aws-kinesisfirehose-s3` and `aws-kinesisstreams-kinesisfirehose-s3` patterns can get `existingLoggingBucketObj` as a prop. If `existingLoggingBucketObj` is provided, the S3 bucket created by the patterns are going to log to `existingLoggingBucketObj`.
4. All pattern construct props and pattern properties are alphabetic order.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-QfXHf1ULVFb1
* Commit ID: 1ed283c8f74e8a349616368481282dc0f08aa7a3
* Result: FAILED
* [Build Logs](https://idenu9lil3.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c5ff03c9-ff8f-48b9-9c3c-b915f82938a6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by beomseoklee** [comment]
It seems like `cfn_nag` added a new rule, and the build failed because of the new rule.

```
| WARN W88
| Kinesis Firehose DeliveryStream of type DirectPut should specify SSE.
```

The warning happens on `aws-kinesisfirehose-s3`, `aws-events-rule-kinesisfirehose-s3`, and `aws-iot-kinesisfirehose-s3` patterns. Do you want me to fix all of them or just `aws-kinesisfirehose-s3` as the pattern is related to this pull request? In addition, how do you want me to fix this? 1) Suppressing the rule or 2) enabling SSE?

**Comment by biffgaut** [comment]
Wait – you’re in github, the fix is in CodeCommit. Lemme talk to Hitendra on Monday to see the best way to correlate these changes – you’re one of a couple PRs in GitHub facing this issue.

Biff


Biff Gaut
biffgaut@

File: source/patterns/@aws-solutions-constructs/aws-kinesisfirehose-s3/test/integ.pre-existing-logging-bucket.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { Bucket, IBucket } from '@aws-cdk/aws-s3';
+import { App, Stack } from '@aws-cdk/core';
+import { KinesisFirehoseToS3 } from '../lib';
+
+// Setup
+const app = new App();
+const stack = new Stack(app, 'test-firehose-s3-pre-existing-logging-bucket-stack');
+stack.templateOptions.description = 'Integration Test for aws-kinesisfirehose-s3';
+
+const myLoggingBucket: IBucket = Bucket.fromBucketName(stack, 'myLoggingBucket', 'myLoggingBucket');
```
[the remark this answers is not in the document]
Comment (beomseoklee) [reply source/patterns/@aws-solutions-constructs/aws-kinesisfirehose-s3/test/integ.pre-existing-logging-bucket.ts:24]: `myLoggingBucket` is an arbitrary bucket to assume the bucket exists, but it seems like you prefer to use the below code, so I'll change the source code:
```typescript
const existingBucket = CreateScrapBucket(stack, {});
const myLoggingBucket: s3.IBucket = s3.Bucket.fromBucketName(stack, 'myLoggingBucket', existingBucket.bucketName);
```

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-QfXHf1ULVFb1
* Commit ID: 2f6a3e98cb5e595fa765943a5c336097addabc75
* Result: SUCCEEDED
* [Build Logs](https://idenu9lil3.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=88e3889d-97ea-4489-87ec-93a416341301%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-kinesisfirehose-s3/lib/index.ts [hunk 2]
```diff
@@ -68,8 +80,24 @@ export class KinesisFirehoseToS3 extends Construct {
...
+      let { bucketProps } = props;
+
+      // Setup logging S3 Bucket
+      if (props.existingLoggingBucketObj) {
+        if (!bucketProps) {
+          bucketProps = {
+            serverAccessLogsBucket: props.existingLoggingBucketObj
+          };
+        } else {
+          bucketProps = {
```
[the remark this answers is not in the document]
Comment (beomseoklee) [reply source/patterns/@aws-solutions-constructs/aws-kinesisfirehose-s3/lib/index.ts]: Done!

File: source/patterns/@aws-solutions-constructs/aws-kinesisfirehose-s3/test/integ.pre-existing-bucket.ts [hunk 3]
```diff
@@ -12,24 +12,23 @@
  */
 
 // Imports
-import { App, Stack } from "@aws-cdk/core";
-import { KinesisFirehoseToS3 } from "../lib";
-import * as s3 from "@aws-cdk/aws-s3";
-import * as cdk from "@aws-cdk/core";
+import { Bucket, IBucket } from '@aws-cdk/aws-s3';
```
[the remark this answers is not in the document]
Comment (beomseoklee) [reply source/patterns/@aws-solutions-constructs/aws-kinesisfirehose-s3/test/integ.pre-existing-bucket.ts]: Done!

File: source/patterns/@aws-solutions-constructs/aws-kinesisfirehose-s3/test/test.kinesisfirehose-s3.test.ts [hunk 4]
```diff
@@ -11,25 +11,24 @@
  *  and limitations under the License.
  */
 
-import { SynthUtils } from '@aws-cdk/assert';
-import { KinesisFirehoseToS3, KinesisFirehoseToS3Props } from "../lib";
-import * as cdk from '@aws-cdk/core';
 import '@aws-cdk/assert/jest';
+import { SynthUtils } from '@aws-cdk/assert';
+import { Bucket, IBucket } from "@aws-cdk/aws-s3";
+import { Stack } from '@aws-cdk/core';
```
[the remark this answers is not in the document]
Comment (beomseoklee) [reply source/patterns/@aws-solutions-constructs/aws-kinesisfirehose-s3/test/test.kinesisfirehose-s3.test.ts]: Done!

File: source/patterns/@aws-solutions-constructs/aws-kinesisstreams-kinesisfirehose-s3/test/test.kinesisfirehose-s3.test.ts [hunk 5]
```diff
@@ -11,30 +11,27 @@
...
-import { SynthUtils } from '@aws-cdk/assert';
-import { KinesisStreamsToKinesisFirehoseToS3, KinesisStreamsToKinesisFirehoseToS3Props } from '../lib';
-import * as cdk from '@aws-cdk/core';
 import '@aws-cdk/assert/jest';
-import { Stack } from '@aws-cdk/core';
-import * as defaults from '@aws-solutions-constructs/core';
+import { SynthUtils } from '@aws-cdk/assert';
 import { Role, ServicePrincipal } from '@aws-cdk/aws-iam';
+import { Bucket, IBucket } from '@aws-cdk/aws-s3';
+import { Stack } from '@aws-cdk/core';
```
[the remark this answers is not in the document]
Comment (beomseoklee) [reply source/patterns/@aws-solutions-constructs/aws-kinesisstreams-kinesisfirehose-s3/test/test.kinesisfirehose-s3.test.ts]: Done!

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-QfXHf1ULVFb1
* Commit ID: 5cee59c7e582c9756ef8a55cf29a19916fc1324d
* Result: SUCCEEDED
* [Build Logs](https://idenu9lil3.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=08439467-aec8-4280-8638-ced31714e2f5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-QfXHf1ULVFb1
* Commit ID: 62e5dca1406e849067a9337e39bb7d68b77ad9b2
* Result: SUCCEEDED
* [Build Logs](https://idenu9lil3.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d2168841-40fd-4c74-813b-8958c3a30b6c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by beomseoklee** [comment]
Merged the latest version, `1.88.0`, build is passed, and changed the PR based on the feedback.
Please let me know if anything more is required.

**Comment by biffgaut** [comment]
Biff Gaut
biffgaut@

**Comment by biffgaut** [comment]
I will review this today – thanks!

Biff Gaut
biffgaut@

**Review by hnishar — APPROVED** [review]

**Pull request merged**